source/develop/release-management/features/virt/engine-dedicated-cpus.md

```diff
@@ -0,0 +1,102 @@
+---
+Title: Engine Dedicated CPUs
+Category: feature
+Authors: sabusale
+---
+
+# Engine Dedicated CPUs
+
+## Summary
+
+Once choosing dedicated CPU policy, The guest vCPUs will be strictly pinned to a set of host pCPUs
+(similarly to static CPU pinning). The set of pCPUs will be chosen to match the required guest CPU topology.
+
+Engine handles the pinning automatically when starting the VM as opposed to the manual pinning that is
+done by the user and is static.
+
+## Owner
+
+-   Name: Saif Abu Saleh (sabusale)
+-   Email: sabusale@redhat.com
+
+## Responsibilities
+
+VDSM:
+
+- Manages the available pinned CPUs
+
+Engine:
+- Schedule VM to run on best satisifing host
+- creates vCpu to pCpu mappings
+
+
+## Backend
+
+1. Add dedicated to CPU Pinning Policy enum
+
+2.  Scheduling
+    1. Start VM:
+        - Scheduling phase
+            - Score all hosts to find best matching host with enough pCPUs
+        - Run phase
+            - Create vCpu to pCpu mapping based on the host free pCPUs reported
+            - Update VM CurrentCpuPinning DB field with the mapping
+
+    2. Stop/Destroy VM:
+        - Clear VM dynamic CPU pinning field
+    
+3. Send vcpupin with cpusets and policy name in VM metadata
+   1. Send the policy name, available policy names: { none, pin, dedicated, siblings, isolate-threads }.
+   2. Send the cpusets (it will be stored in VM_Dynamic)
+
+4. Migrate VDSM call
+   1. Calculate vCPU to pCPU mapping based on the scheduled host
+   2. Send cpusets additional parameter
+   
+5. setNumberOfCpus VDSM call
+   1. Send cpusets additional parameter
+
+
+## Database
+
+* Save CPU mappings in field CurrentCpuPinning field in vm_dynamic table
+
+### Example
+2 hosts are running and reporting the following shared CPUs
+
+host A: CPUs 1, 2 on socket 1 and 3, 4 on socket 2
+
+host B: CPUs 1 on socket 1 and 4 on socket 2
```

Comment:
I don't see why we care about this information from the host
let's take host B for instance,
assuming pCPU 0 is reported as a `reserved cpu` and we know that pCPU 2, 3, 5-7 are exclusively pinned to some vCPUs, the engine can know that pCPUs 1 and 4 are not exclusively pinned - the engine would anyway need to know which vCPU is pinned to which pCPU so with that information, together with the reserved cpu, it has all the data to have a clear picture of what happens on the host, right?